smqtk_detection/impls/detect_image_objects/centernet.py

```diff
@@ -565,8 +574,25 @@ def _gather_feat(feat, ind):  # type: ignore
 
         return feat
 
+    def _gather_feat_mps(feat, ind):  # type: ignore
```

Comment:
Instead of this method specialized for MPS, could we implement the workaround in a `torch.gather` wrapper?

```python
def _gather(input: torch.Tensor, dim: int, index: torch.Tensor) -> torch.Tensor:
    if input.device.type != "mps":
        return torch.gather(input, dim, index)

    ...
```

This way, if the bug is ever fixed upstream, we only need to remove the workaround in a single instead of multiple places. Plus, that would avoid branching on MPS in multiple places as well.

Reply:
That makes much more sense, a far nicer pattern! I'll implement this now.